### PR TITLE
No duplicate variable rule no longer walks type literals

### PR DESCRIPTION
**Pull request by gscshoyru**

### CODE REVIEW

**Comment by adidahiya** [comment]
:+1: lgtm

File: test/files/rules/duplicate-variable.test.ts [hunk 1]
```diff
@@ -103,3 +103,6 @@ function testArguments2(x: number, y: number): void {
     var x = 1;
     let y = 2;
 }
+
+var references: {[vertex: string]: any};
```
Comment (ashwinr) [inline test/files/rules/duplicate-variable.test.ts]: also add a test for parameters:

```
function blah(arg1: {[key: string]: any, arg2: {[key:string]: any}) {
}
```

Comment (gscshoyru) [inline test/files/rules/duplicate-variable.test.ts]: Done.

**Comment by ashwinr** [comment]
this is not fully fixed. here's an exception that's still causing false positives:

```
    export interface IClipboard {
        copy(key: string, state: any): void;
        paste(key: string): any;
    }
```

duplicate "key"

**Comment by ashwinr** [comment]
another one:

`findMaxOrMin(values: any[], defaultValue: number, operation: (...values: any[]) => number)`

duplicate "values"

**Comment by ashwinr** [comment]
one more:

```
try {
    //
} catch (e) {
    e.blah();
    //
}

try {
    //
} catch (e) {
    e.blah();
    //
}
```

duplicate "e"

**Comment by adidahiya** [comment]
made a new issue for these false positives: #372

**Pull request merged**